src/main/java/rx/internal/operators/OperatorDoOnUnsubscribe.java

```diff
@@ -41,22 +42,6 @@ public OperatorDoOnUnsubscribe(Action0 unsubscribe) {
 
         // Pass through since this operator is for notification only, there is
         // no change to the stream whatsoever.
-        return new Subscriber<T>(child) {
-            @Override
-            public void onStart() {
-            }
-            @Override
-            public void onNext(T t) {
-                child.onNext(t);
-            }
-            @Override
-            public void onError(Throwable e) {
-                child.onError(e);
-            }
-            @Override
-            public void onCompleted() {
-                child.onCompleted();
-            }
-        };
+        return Subscribers.wrap(child);
```

Comment:
Same here. The origin of the issue is in `lift`: https://github.com/ReactiveX/RxJava/blob/1.x/src/main/java/rx/Observable.java#L144